core/src/test/java/org/apache/iceberg/rest/TestResourcePaths.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest;
+
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestResourcePaths {
+  private final String prefix = "ws/catalog";
+  private final ResourcePaths withPrefix = ResourcePaths.forCatalogProperties(ImmutableMap.of("prefix", prefix));
+  private final ResourcePaths withoutPrefix = ResourcePaths.forCatalogProperties(ImmutableMap.of());
+
+  @Test
+  public void testConfigPath() {
+    // prefix does not affect the config route because config is merged into catalog properties
+    Assert.assertEquals(ResourcePaths.config(), "v1/config");
+  }
+
+  @Test
+  public void testNamespaces() {
+    Assert.assertEquals("v1/ws/catalog/namespaces", withPrefix.namespaces());
+    Assert.assertEquals("v1/namespaces", withoutPrefix.namespaces());
+  }
+
+  @Test
+  public void testNamespace() {
+    Namespace ns = Namespace.of("ns");
+    Assert.assertEquals("v1/ws/catalog/namespaces/ns", withPrefix.namespace(ns));
+    Assert.assertEquals("v1/namespaces/ns", withoutPrefix.namespace(ns));
+  }
+
+  @Test
+  public void testNamespaceWithSlash() {
+    Namespace ns = Namespace.of("n/s");
+    Assert.assertEquals("v1/ws/catalog/namespaces/n%2Fs", withPrefix.namespace(ns));
+    Assert.assertEquals("v1/namespaces/n%2Fs", withoutPrefix.namespace(ns));
+  }
+
+  @Test
+  public void testNamespaceWithMultipartNamespace() {
+    Namespace ns = Namespace.of("n", "s");
+    Assert.assertEquals("v1/ws/catalog/namespaces/n%00s", withPrefix.namespace(ns));
+    Assert.assertEquals("v1/namespaces/n%00s", withoutPrefix.namespace(ns));
```

Comment:
@kbendick, these tests failed at first and I had to update the null byte joiner. I think the problem was that the URL encoder was running on each namespace part and not on the whole thing, so null bytes were left unencoded.

Reply:
Ah ok good catch. Thank you for fixing it!